File: Java/Binary_search.java
public class Binary_search {
    public static void main(String[] args) {
        int[] arr={-2,0,1,5,7,12,18,22,24,34};
        int target=18;
        int ans=binarysearch(arr, target);
        System.out.println("target ele. index is=" +  ans);
    }

    static int binarysearch(int[] arr, int target){
        int start=0;
        int end=arr.length-1;

        while(start<=end){
            int mid=start + (end-start) / 2;

            if(target<arr[mid]){
                end=mid-1;
            }
            else if(target>arr[mid]){
                start=mid+1;
            }
            else{
                return mid;
            }

        }

        return -1;

    }
    
}

File: C++/binarySearch.cpp
#include <iostream>

using namespace std;

int binarySearch(int arr[], int siz, int key)
{
    int start = 0;
    int ending = siz - 1;
    //int mid = (start + ending)/2;
    int mid = start +(ending - start)/2;

    while(start <= ending)
    {
        if (arr[mid] == key)
            return mid;
        if (key > arr[mid])
            start = mid + 1;
        else
            ending = mid - 1;

        //mid = (start + ending)/2;
        mid = start + (ending - start)/2;
    }
    return -1;
}

int main()
{
    int even[6] = {1,3,4,5,6,8};
    int odd[5] = {4,5,6,7,8};

    int evenIndex = binarySearch(even, 6, 6);
    cout << "Index of 6 is: " << evenIndex <<endl;

    int OddIndex = binarySearch(odd, 5, 4);
    cout << "Index of 4 in odd is: " <<OddIndex << endl;


}
